### PR TITLE
Updates FormControl validation example to better align with the interface guidelines

### PR DESCRIPTION
**Pull request by mperrotti**
I added a more dynamic validation example that conforms to the form validation patterns defined in the form interface guidelines.

### CODE REVIEW

File: docs/content/FormControl.mdx [hunk 1]
```diff
@@ -284,41 +284,150 @@ We encourage using `FormControl` alongside all standalone form components like [
...
+  }
+
+  if (/\s/g.test(value)) {
+    validationKeys.push('handleHasSpaces')
+  }
+
+  return validationKeys
+}
+
+const ValidationOnSubmitExample = () => {
```
Comment (mperrotti) [inline docs/content/FormControl.mdx:319]: Help wanted: this covers the functionality I wanted to showcase, but there's a lot of code to look through. I could use some help simplifying the code for this example.

**Comment by lesliecdubs** [comment]
👋🏻 @mperrotti would you still like reviews on this? Happy to get a buddy assigned if you don't want this to go stale.

**Comment by mperrotti** [comment]
@lesliecdubs - I completely forgot about this 😅 

We decided to put this example into the stories instead of the docs site. I just got distracted by other work and haven't come back to it yet.

**Comment by mperrotti** [comment]
I made the update, but after talking to @ericwbailey, I'm uncertain that we'll stick with [the guidance](https://primer.style/design/ui-patterns/forms#validation-on-submit) that this change is demonstrating:
> If an interactive summary of errors is not shown, the first invalid input should be focused and scrolled into the viewport

I'll be bringing it to next week's a11y design office hours. Then I'll either mark this PR as "Ready for review" or close it.

**Comment by mperrotti** [comment]
When I brought this to a11y office hours, we decided to stick with the existing guidance. I will come back to this later this week or early next week.

**Pull request closed without merging**